packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebViewClient.java

```diff
@@ -30,6 +30,7 @@
   private static final String TAG = "FlutterWebViewClient";
   private final MethodChannel methodChannel;
   private boolean hasNavigationDelegate;
+  private boolean hasProgressTracking;
```

Comment:
Good catch, this flag wasn't used properly. I've updated the PR so that progress changes are sent to the method channel only if `hasProgressTracking` is `true`.